### PR TITLE
MONTH and YEAR functions

### PR DESCRIPTION
**Pull request by AntonArsentiev**

### CODE REVIEW

File: efc/rpn_builder/parser/functions.py [hunk 1]
```diff
@@ -681,6 +683,22 @@ def year_frac(dt1, dt2, tp=None):
...
 
+YMD_DATE_FORMAT = '%Y{0}%m{0}%d'
+DMY_DATE_FORMAT = '%d{0}%m{0}%Y'
+
+
+def date_func(key, cell_operand):
+    for date_format in (YMD_DATE_FORMAT, DMY_DATE_FORMAT):
+        for separator in '.-/':
+            try:
+                value = datetime.strptime(cell_operand.string, date_format.format(separator))
```
Comment (ulalka) [inline efc/rpn_builder/parser/functions.py:694]: `format` method pretty slow, use `%` instead

File: efc/rpn_builder/parser/functions.py [hunk 2]
```diff
@@ -681,6 +683,22 @@ def year_frac(dt1, dt2, tp=None):
...
+                return getattr(value, key)
+            except:
```
Comment (ulalka) [inline efc/rpn_builder/parser/functions.py:696]: `except` without an exception class handle every exception, e.g. `KeyboardInterrupt` or `SystemExit`. You should specify the exception class.

File: efc/rpn_builder/parser/functions.py [hunk 3]
```diff
@@ -681,6 +683,22 @@ def year_frac(dt1, dt2, tp=None):
     return v
 
 
+YMD_DATE_FORMAT = '%Y{0}%m{0}%d'
+DMY_DATE_FORMAT = '%d{0}%m{0}%Y'
+
+
+def date_func(key, cell_operand):
+    for date_format in (YMD_DATE_FORMAT, DMY_DATE_FORMAT):
```
Comment (ulalka) [inline efc/rpn_builder/parser/functions.py:691]: According to the documentation, there are correct formats:
`=MONTH("28.02.2011")`
`=MONTH("28/02/2011")`
`=MONTH("28-02-2011")` 
`=MONTH("28-Feb-2011")`
`=MONTH("28-Фев-2011")`
`=MONTH("28-february-2011")`
`=MONTH("28 february 2011")`
`=MONTH("28 февраль 2011")`
`=MONTH("2011.02.28")` 
`=MONTH("2011-02-28")`
`=MONTH(40602)`

You should support all of them

File: tests/test_rpn/test_functions.py [hunk 4]
```diff
@@ -310,3 +312,21 @@ def test_YEARFRAC(calc):
     assert calc('YEARFRAC(43405, 43465, 4)', 'Yet another sheet').value == 59 / 360
     assert calc('YEARFRAC(43889, 43890, 4)', 'Yet another sheet').value == 1 / 360
     assert calc('YEARFRAC(43889, 43891, 4)', 'Yet another sheet').value == 3 / 360
+
+
+def test_MONTH_YEAR(calc):
+    dt = date(1994, 2, 18)
+
+    for excel_function in ('MONTH', 'YEAR'):
```
Comment (ulalka) [inline tests/test_rpn/test_functions.py:320]: omg, use `pytest.mark.parametrize`

File: efc/rpn_builder/parser/functions.py (hunk 2 above)
Comment (ulalka) [reply efc/rpn_builder/parser/functions.py:696]: Handle exception may be slower than using `if` statement. Please, check it and use faster solution.

**Review by ulalka — CHANGES_REQUESTED** [review]

**Comment by AntonArsentiev** [comment]
closing pull-request without comments

**Pull request closed without merging**